nengo/utils/tests/test_socket.py

```diff
@@ -0,0 +1,151 @@
+import math
+import threading
+import numpy as np
+
+import nengo
+from nengo.utils import sockets
+
+
+class SimThread(threading.Thread):
+    def __init__(self, simulator, sim_time, lock=None, wait=False):
+        threading.Thread.__init__(self)
+        self.sim = simulator
+        self.sim_time = sim_time
+        self.lock = lock
+        self.wait = wait
+
+    def run(self):
+        if self.lock is not None and self.wait:
+            self.lock.wait()
+        if self.lock is not None and not self.wait:
+            self.lock.set()
+        self.sim.run(self.sim_time)
+
+
+def test_send_recv_chain(Simulator, plt, seed, rng):
+    # Model that sends data
+    udp_send = sockets.UDPSocket(dest_port=54321, send_dim=1)
+    m_send = nengo.Network(label='Send', seed=seed)
+    with m_send:
+        input = nengo.Node(output=lambda t: math.sin(10 * t))
```

Comment:
This might be crazy, but for code simplicity, why not use `np.sin` and thus not need to `import math` in the first place?